src/audio/pcm_converter/pcm_converter.c

```diff
@@ -33,9 +33,9 @@ int pcm_convert_as_linear(const struct audio_stream __sparse_cache *source, uint
 
 	/* assert enough avail/free samples in source and sink buffer */
 	if (audio_stream_get_avail_samples(source) < samples + ioffset)
-		return -EINVAL;
+		return;
 	if (audio_stream_get_free_samples(sink) < samples + ooffset)
-		return -EINVAL;
+		return;
```

Comment:
this loses these potential errors, but yes, I can see, that possibly they aren't needed

Reply:
yeah, I did not remove it totally, we can try to remove those checks later.
SOF worry too much about each possible error and this cause much more overhead in component.

Reply:
@lyakh , I intend to remove this check, as other converter function does not have this check, what do you think？